CMakeLists.txt

```diff
@@ -1,49 +1,107 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.19)
```

Comment:
`COMPATIBILITY_MODE` of [write_basic_package_version_file](https://cmake.org/cmake/help/latest/module/CMakePackageConfigHelpers.html#generating-a-package-version-file) requires 3.19. Hope this isn't too recent.

Reply:
CMake 3.19 isn't so recent I'd expect issues but I also do not use CMake much myself (as may be apparent 😅). I see in the docs there's both `COMPATIBILITY` and `COMPATIBILITY_MODE` referenced, the first for before 3.19. I'm not sure of the difference. Maybe we could get away with `COMPATIBILITY` and be able to relax the version requirement?

Reply:
`COMPATIBILITY_MODE` refers to the values passed to the argument-keyword `COMPATIBILITY`. That being said I misread the documentation, what they're saying is that the behavior of those changed not that they were introduced. I'm unsure what would be the lowest working version, maybe 3.1 is still fine.

Reply:
Yeah, I think we're saying the same thing. As I said in my second comment, I initially misread the documentation. The PR was merged before I could double check which exact version is the lowest usable and which policy version could/should be used. I'd venture to guess that `3.1...3.24` is fine.

Regarding visibility, it was outside the scope of the PR but I agree, that being said it can be set through `<LANG>_VISIBILITY_PRESET`. I would add a CMakePresets.json file rather than modifying the CMakeLists.txt for that purpose.